wordpress-mobile/WordPress-iOS
# Feature/6358 related posts ui

Refs #6358 

This is the fourth and finally development PR for adding related posts to the ReaderDetailViewController.  We'll follow up this PR with a final design review before merging to develop.  Since this PR includes model changes, we'll want to try to wrap it up before cutting release/7.1. 

In this PR the RelatedPostsViewController is fleshed out.  Related posts are shown in two sections. One for posts that are from the same blog as the post currently shown in the detail, and one for posts that are from other WordPress.com sites.  Tapping a related post shows that post in a new detail.  Posts from other WPCom sites should display a follow button. 

To test:
View the reader detail for a wpcom post.  Confirm that related posts are loaded and displayed below the post content.
- Confirm the UI updates correctly when changing orientation. 
- Tap to view a related post. Confirm the detail loads and the correct post is shown. 
- Tap to follow a site via a related WPCom post. Tap to unfollow.
- Confirm that the action buttons (likes, comments, etc.) are not shown on the related post cards.
- Confirm that a longpress gesture on the related post cards does not invoke text selection. 

For paranoia, please check posts in a few different streams.  I noticed some layout issues in the post cards (squashed labels) after moving the card out of the cell and into its own view.  It should be resolved with this PR but just in case...

Needs review: @kurzee could I trouble you with this one? 

Examples:

![simulator screen shot feb 20 2017 11 19 18 am](https://cloud.githubusercontent.com/assets/1435271/23135503/95a0339e-f75e-11e6-848d-bb828f9615ef.png)
![simulator screen shot feb 20 2017 11 19 31 am](https://cloud.githubusercontent.com/assets/1435271/23135512/9813242e-f75e-11e6-89b4-4a1ac0aab551.png)
